src/api/app/models/event_subscription/generate_hash_for_subscriber.rb

```diff
@@ -0,0 +1,64 @@
+class EventSubscription
```

Comment:
Perhaps the fact that the subscriptions are grouped by event_class is something that is only used by the view but the rest is definitely business logic / domain layer stuff (as opposed to presentation layer). And I have to group them by event_class anyway in order to make the right queries so I still really dont think this belongs in a helper.

This class models this behaviour: "if defaults are set and the user has no notification settings then use the default setting. if the defaults are set but the user does have their own notification settings set then use the user's settings. otherwise the notifications are disabled." If thats not business I dont know what is?

And yes this is querying data into a specific structure - exactly what the models classes are for, not helpers.

A helper could be used for something like "if the notification is set to receive then the colour of the text should be green".

Here are some discussions and articles about a similar situations:

And also some other stuff about helpers vs models:
https://www.reddit.com/r/rails/comments/27s3jw/models_vs_helpers/
https://www.toptal.com/ruby-on-rails/top-10-mistakes-that-rails-programmers-make
https://stackoverflow.com/questions/5019794/when-to-use-helpers-vs-model
https://coderwall.com/p/jx9tca/when-i-use-helpers-partials-presenters-and-decorators